c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1627,6 +1774,7 @@ class THCCachingAllocator {
       device_allocator.resize(device_count);
       for (const auto i : c10::irange(size, device_count)) {
         device_allocator[i] = std::make_unique<DeviceCachingAllocator>();
+        memory_tracker.alloc_free_events[i].resize(1);
```

Comment:
why do you need `resize(1)` here?

Reply:
I removed it and edited the append function. Thank you.